src/components/cc-logs-instances/cc-logs-instances.js

```diff
@@ -0,0 +1,567 @@
+   * @param {Deployment} deployment
+   */
+  _renderDeploymentDetails (deployment) {
+    // todo: handle unknown deployment !
+    return html`
+      <div class="deployment">
+        <div>${this._renderDeploymentState(deployment)}</div>
+        <div>${deployment.creationDate != null ? html`
+          ${i18n('cc-logs-instances.deployment.deployed')}
+          <cc-datetime-relative datetime=${deployment.creationDate.toISOString()}></cc-datetime-relative>
+        ` : 'Unknown deployment'}
+        </div>
+        ${this._renderCommit(deployment.commitId)}
+      </div>
+    `;
+  }
+
+  /**
+   * @param {Deployment} deployment
+   */
+  _renderDeploymentState (deployment) {
+    if (deployment.state === 'WIP') {
+      return html`
+        <cc-loader accessible-name="${i18n('cc-logs-instances.deployment.state.wip')}"
+                   class="deployment_state"></cc-loader>`;
+    }
+
+    if (deployment.state === 'OK') {
+      return html`
+        <cc-icon .icon=${iconDeploymentOk}
+                 accessible-name="${i18n('cc-logs-instances.deployment.state.ok')}"
+                 class="deployment_state deployment_state--ok"></cc-icon>`;
+    }
+
+    if (deployment.state === 'FAILED') {
+      return html`
+        <cc-icon .icon=${iconDeploymentFailed}
+                 accessible-name="${i18n('cc-logs-instances.deployment.state.failed')}"
+                 class="deployment_state deployment_state--failed"></cc-icon>`;
+    }
+
+    if (deployment.state === 'CANCELLED') {
+      return html`
+        <cc-icon .icon=${iconDeploymentCancelled}
+                 accessible-name="${i18n('cc-logs-instances.deployment.state.cancelled')}"
+                 class="deployment_state deployment_state--cancelled"></cc-icon>`;
+    }
+
+    return '';
+  }
+
+  /**
+   *
+   * @param {InstanceState} state
+   */
+  _renderInstanceState (state) {
+    const getAccessibleName = () => {
+      if (state === 'BOOTING') {
+        return i18n('cc-logs-instances.instance.state.booting');
+      }
+      if (state === 'STARTING') {
+        return i18n('cc-logs-instances.instance.state.starting');
+      }
+      if (state === 'DEPLOYING') {
+        return i18n('cc-logs-instances.instance.state.deploying');
+      }
+      if (state === 'READY') {
+        return i18n('cc-logs-instances.instance.state.ready');
+      }
+      if (state === 'UP') {
+        return i18n('cc-logs-instances.instance.state.up');
+      }
+      if (state === 'STOPPING') {
+        return i18n('cc-logs-instances.instance.state.stopping');
+      }
+      if (state === 'DELETED') {
+        return i18n('cc-logs-instances.instance.state.deleted');
+      }
+    };
+
+    const accessibleName = getAccessibleName(state);
+
+    if (state === 'BOOTING' || state === 'STARTING' || state === 'DEPLOYING') {
+      return html`
+        <cc-loader accessible-name="${accessibleName}" 
+                   class="instance_state"></cc-loader>`;
+    }
+
+    if (state === 'READY') {
```

Comment:
thought: maybe we could have avoided these cascade ifs. 
If we:
* had a function to retrieve the icon with the state given
* Either have the state lowercased to put dinamycally the class.
* Or have the class name in the CSS uppercased.

Anyway, that's just a thought, maybe I'm wrong and there's something I am not aware of, don't worry about it. :wink:

example:

```js
return html`
  <!-- Icon could have been retrieved with a dedicated function -->
  <cc-icon .icon=${getIconState(state)}
                 accessible-name="${accessibleName}"
                 class="instance_state instance_state--${state.toLowerCase()}"></cc-icon>`;
`;
```

or if the class was in uppercase

```js

return html`
  <!-- Icon could have been retrieved with a dedicated function -->
  <cc-icon .icon=${getIconState(state)}
                 accessible-name="${accessibleName}"
                 class="instance_state instance_state--${state}"></cc-icon>`;
`;
```